Model/src/BoardPackage/JailSquare.java

```diff
@@ -1,7 +1,83 @@
 package BoardPackage;
 
-/**
- * Created by Dylan Kernohan on 2/10/18.
- */
-public class JailSquare {
+import GamePackage.Player;
+
+import java.util.ArrayList;
+
+/*********************************************************************
+ * The JailSquare Class tracks what players are in jail.
+ *
+ * @author David Baas
+ * @version 2/14/2018
+ ********************************************************************/
+public class JailSquare extends BoardSquare {
+
+    /**An ArrayList to store what players are in jail.*/
+    private ArrayList<Player> inJail;
+
+    /*****************************************************************
+     * The default constructor creates a JailSquare setting its name
+     * variable and initializes the inJail ArrayList.
+     *
+     * @param name sets the name of the board square.
+     ****************************************************************/
+    public JailSquare(String name){
+
+        super(name);
+
+        inJail = new ArrayList<Player>();
+    }
+
+    /*****************************************************************
+     * The addToJail method adds a single player into the jail list
+     * There is a check when this method is invoked that the same
+     * player is not added to jail twice.
+     *
+     * @param p the Player who is being sent to jail.
+     ****************************************************************/
+    public void addToJail(Player p){
+        for(Player o: inJail){
+            if(p.equals(o))
+                return;
+        }
+        inJail.add(p);
+    }
+
+    /*****************************************************************
+     * The removeFrom Jail method removes a player from being in jail.
+     * There is a distinction from being in jail and visiting jail.
+     * There is a check to see if the player being removed from jail
+     * is within the inJail list.
+     *
+     * @param p the Player being removed from jail.
+     * @return true if the Player specified exist in the inJail list
+     * and is removed, false otherwise.
+     ****************************************************************/
+    public boolean removeFromJail(Player p){
+        if(inJail.contains(p)) {

```

Comment:
this method is very redundant, since the ArrayList.remove(Object o) already returns a boolean value. you could just do: return inJail.remove(player);

Reply:
Agreed.

Reply:
Make sure you leave the @return with description though